ui/app/pages/settings/plugins-tab/plugins-list/index.js

```diff
@@ -0,0 +1,163 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import deepEqual from 'fast-deep-equal'
+import Button from '../../../../components/ui/button'
+
+export default class PluginsList extends Component {
+
+  static propTypes = {
+    plugins: PropTypes.object.isRequired,
+    removePlugins: PropTypes.func.isRequired,
+  }
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      plugins: {
+        ...this.getPluginsState(props),
+        // [pluginName]: {
+        //   selected: boolean,
+        // },
+      },
+    }
+  }
+
+  componentDidUpdate () {
+    const newPluginState = this.getPluginsState(this.props)
+    // if plugins have been added or removed, reset state
+    // TODO:review is this unnecessary given React's state management?
+    // Like, will React not re-render anyway if state isn't different?
+    if (!deepEqual(
+      Object.keys(this.state.plugins), Object.keys(newPluginState))
+    ) {
+      this.setState({ plugins: newPluginState })
+    }
+  }
+
+  getPluginsState (props) {
+    const { plugins } = props
+    return Object.keys(plugins).reduce((acc, pluginName) => {
+      acc[pluginName] = {
+        selected: true,
+      }
+      return acc
+    }, {})
+  }
+
+  onPluginToggle = id => () => {
```

Comment:
This line is a little weird for me to read. Maybe it's just me. I guess this means `onPluginToggle` is a function that takes an `id` and returns a function that takes no parameters? I think that's complex enough to include some parens and curly braces for clarity, like:

```
onPluginToggle: (id) => { return () => {
```

Or maybe it should match the way other methods are defined here?

```
onPluginToggle (id) {
  return () => {
```

Reply:
Added parens and an explanatory comment